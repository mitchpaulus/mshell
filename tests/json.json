{
  "string": "hello",
  "number": 42.5,
  "integer": 10,
  "boolean_true": true,
  "boolean_false": false,
  "null_value": null,
  "array": [1, "two", null, false, {"nested": "object"}],
  "object": {
    "nested_string": "world",
    "nested_array": [3, 4, 5]
  }
}
