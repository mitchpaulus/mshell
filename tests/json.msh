'[1, 2, 3]' parseJson str wl

'{
    "name": "John",
    "age": 30,
    "city": "New York",
    "hasPets": true,
    "isBad": null
}' parseJson str wl

`json.json` parseJson str wl
