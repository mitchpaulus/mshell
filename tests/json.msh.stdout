[1 2 3]
{"age": {"type": "Float", "value": 30.000000}, "city": {"type": "String", "content": "New York"}, "hasPets": {"type": "Boolean", "value": true}, "isBad": {"type": "Maybe", "value": null}, "name": {"type": "String", "content": "John"}}
{"array": {"type": "List", "items": [{"type": "Float", "value": 1.000000}, {"type": "String", "content": "two"}, {"type": "Maybe", "value": null}, {"type": "Boolean", "value": false}, {"nested": {"type": "String", "content": "object"}}]}, "boolean_false": {"type": "Boolean", "value": false}, "boolean_true": {"type": "Boolean", "value": true}, "integer": {"type": "Float", "value": 10.000000}, "null_value": {"type": "Maybe", "value": null}, "number": {"type": "Float", "value": 42.500000}, "object": {"nested_array": {"type": "List", "items": [{"type": "Float", "value": 3.000000}, {"type": "Float", "value": 4.000000}, {"type": "Float", "value": 5.000000}]}, "nested_string": {"type": "String", "content": "world"}}, "string": {"type": "String", "content": "hello"}}
